File: Python/BOJ/BOJ11047.py
n, k = map(int, input().split())
a = []
for _ in range(n):
    a.append(int(input()))

count = 0

for i in range(n):
    if a[i] > k or i == n - 1:
        coin = i
        break

while True:
    if k == 0:
        print(count)
        break
    if k // a[coin] > 0:
        count += k // a[coin]
        k = k % a[coin]
    else:
        coin -= 1

File: Swift/BOJ/BOJ11047.swift
// https://www.acmicpc.net/problem/11047

let nk = readLine()!.split(separator: " ").compactMap { Int($0) }
let n = nk[0]
var k = nk[1]
var a = [Int]()
for _ in 0..<n {
    a.append(Int(readLine()!)!)
}
var answer = 0
for coin in a.reversed() where k > 0 {
    guard coin <= k else { continue }
    answer += k / coin
    k %= coin
}
print(answer)
